File: android/app/src/main/java/com/izzatismail/midtrim/domain/usecase/FetchEntitlementStatusUseCase.kt
package com.izzatismail.midtrim.domain.usecase

import com.izzatismail.midtrim.domain.repository.EntitlementCacheReader

class FetchEntitlementStatusUseCase(
    private val cache: EntitlementCacheReader
) {
    val isPaidUser: Boolean get() = cache.isPurchased
}

File: ios/MidTrim/MidTrim/Domain/UseCases/FetchEntitlementStatusUseCase.swift
import Foundation

struct FetchEntitlementStatusUseCase: @unchecked Sendable {
    private let cache: EntitlementCacheProtocol

    init(cache: EntitlementCacheProtocol) {
        self.cache = cache
    }

    var isPaidUser: Bool {
        get async { await cache.isPurchased }
    }
}
